### PR TITLE
fix: [CDS-77516]: Adds readonly support for azure artifact runtime fields

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

Readonly support has been added for azure artifacts runtime fields

#### Screenshots

After fix -->
Screenshot 2023-08-22 at 1 35 24 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by pankajsingh-harness — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import * as artifactSourceUtils from '../../artifactSourceUtils'
+import { KubernetesSidecarArtifacts } from '../../../KubernetesArtifacts/KubernetesSidecarArtifacts/KubernetesSidecarArtifacts'
+
+import { template, artifacts, projectsMockResponse, feedsMockResponse, packagesMockResponse } from './mocks'
+
+const refetchFeedsMock = jest.fn()
+const refetchProjectsMock = jest.fn()
+
+jest.mock('@common/hooks/useMutateAsGet', () => ({
+  useMutateAsGet: jest.fn().mockImplementation(fn => {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx:26]: Ideally, we should not do this.

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    refetch: jest.fn(),
+    error: null
+  }))
+}))
+
+jest.spyOn(artifactSourceUtils, 'fromPipelineInputTriggerTab')
+jest.spyOn(artifactSourceUtils, 'isFieldfromTriggerTabDisabled')
+jest.spyOn(artifactSourceUtils, 'resetTags').mockImplementation(() => jest.fn())
+
+jest.mock('@connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField', () => ({
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx:62]: We should not do this.

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  ...(jest.requireActual('@connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField') as any),
+  // eslint-disable-next-line react/display-name
+  FormMultiTypeConnectorField: (props: any) => {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx:65]: Mocking component would be bad practice. Only data should be mocked.

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx (hunk 2 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx:62]: Agreed, fixed now

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx (hunk 3 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx:65]: fixed

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AzureArtifactsSource/__tests__/AzureArtifactsSource.test.tsx:26]: This has been done to ensure the mocked value gets picked. Also this hook has separate coverage at the platform level.
This mock implementation enables us to return the desired data via mocks

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**